### PR TITLE
Child merge back office and notification branch

### PR DESCRIPTION
**Pull request by ritesh52n**

### CODE REVIEW

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Model/Common/UiPageTypeModel.cs [hunk 1]
```diff
@@ -12,5 +12,26 @@ public class UiPageTypeModel : Entity
...
+        /// </summary>
+        public int UiNavigationCategoryId { get; set; }
+        /// <summary>
+        /// It Contains The Name For The Ui Navigation Category
+        /// </summary>
+        public string UiNavigationCategoryName { get; set; }
+        /// <summary>
+        /// It Contains The Url With Ui Page Type Id
+        /// </summary>
+        public string FormatedUrl { get; set; }
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Model/Common/UiPageTypeModel.cs:30]: do we need this property, url should have been enough.

Comment (rajguptaH) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Model/Common/UiPageTypeModel.cs:30]: no need of this thing i just remove that

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/wwwroot/js/Notification.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
+﻿//variables to manipulate bell Icon Content
+
+var count = 1;
+var i = 0;
+var mess = " ";
+const z = [];
+//To hide or display the BellIcon
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/wwwroot/js/Notification.js:7]: Give proper naming convention to all the variables. what is x,i, mess, z etc.

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/UiPageValidation/Create.cshtml [hunk 3]
```diff
@@ -1,17 +1,14 @@
...
-
 <h1>Ui Page Validation </h1>
-
 <h4>Create</h4>
 <hr />
 <div class="row">
     <div class="col-md-4">
         <form asp-action="Create" _lpchecked="1" class="mb-2" id="valForm" method="post" asp-controller="UiPageValidation" novalidate>
-            <div asp-validation-summary="All"></div>
+            <div asp-validation-summary="All" ></div>  @*id is added for notification*@
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/UiPageValidation/Create.cshtml:11]: is the commented code is required?

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/Common/Edit.cshtml [hunk 4]
```diff
@@ -111,12 +111,14 @@
...
+
+                document.getElementById(newErrorMessage.fid).style.border = "2px solid red";
                   document.getElementById("lefid").value = newErrorMessage.sourceId;
             }
         });
     }
-</script>
\ No newline at end of file
+</script>
+<script src="~/js/Notification.js"></script> @* Notification function include*@
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/Common/Edit.cshtml:124]: why Notification.js is attached at many pages.

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/Common/Create.cshtml [hunk 5]
```diff
@@ -236,3 +249,4 @@
         });
     }
 </script>
+<script src="~/js/Notification.js"></script> @* Notification function include*@
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/Common/Create.cshtml:252]: again Notification.js included.

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Startup.cs [hunk 6]
```diff
@@ -37,7 +37,7 @@ public void ConfigureServices(IServiceCollection services)
             services.AddSession();
             services.AddControllersWithViews();
             services.AddAutoMapper(typeof(Startup));
-            services.AddControllers();
+            services.AddControllers().AddNewtonsoftJson();
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Startup.cs:40]: what is the use of this AddNewtonsoftJson here?

**Review by smartdusttechnologies — COMMENTED** [review]
1. why there are changes in TestDetails.css as you already have Notification.css file.
2. Remove control specific css and css on build-in classes in Notification.css file.
3. does appsetting.json has changes in ur feature?
4. is this file needed to be checked-in TestingAndCalibrationLabs.Web.UI.csproj.user?
5.

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Startup.cs (hunk 6 above)
Comment (rajguptaH) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Startup.cs:40]: i am passing response from controller to ajax response and for that thing i need to use this library

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/Common/Create.cshtml (hunk 5 above)
Comment (ritesh52n) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/Common/Create.cshtml:252]: removed

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/Common/Edit.cshtml (hunk 4 above)
Comment (ritesh52n) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/Common/Edit.cshtml:124]: removed

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/UiPageValidation/Create.cshtml [hunk 7]
```diff
@@ -45,21 +40,17 @@
...
             </div>
         </form>
     </div>
 </div>
-
 <div>
     <a asp-action="Index">Back to List</a>
 </div>
-
+  <script src="~/js/Notification.js"></script>
```
[the remark this answers is not in the document]
Comment (ritesh52n) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/UiPageValidation/Create.cshtml:53]: yes i have fixed that

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/wwwroot/js/Notification.js (hunk 2 above)
Comment (ritesh52n) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/wwwroot/js/Notification.js:7]: yes i have done that

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/wwwroot/css/Notification.css [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+}
+@media (min-width: 751px) and (max-width: 1020px) {
+
+    .dropdown_menu {
+        margin: 10px;
+        margin-left: 150px;
+    }
+}
+
+.dropdown_menu {
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/wwwroot/css/Notification.css:158]: this is a very generic name. please name it such a way that it cannot conflict

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/UiPageValidation/Create.cshtml (hunk 7 above)
[the remark this answers is not in the document]
Comment (smartdusttechnologies) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/UiPageValidation/Create.cshtml:53]: it is not fixed

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Model/Common/UiPageTypeModel.cs (hunk 1 above)
Comment (smartdusttechnologies) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Model/Common/UiPageTypeModel.cs:30]: it is still there.

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/UiPageValidation/Edit.cshtml [hunk 9]
```diff
@@ -13,7 +13,7 @@
         <form asp-action="Edit" _lpchecked="1" class="mb-2" id="valForm" method="post" asp-controller="UiPageValidation" novalidate>
             <div asp-validation-summary="All"></div>
             <div class="row ms-1 me-1">
-                <div class="col-lg-8  col-sm-12 col-xs-12 mt-3 form-element right-pd-input">
+                <div class="col-lg-8  col-sm-12 col-xs-12 mt-3 form-element-for-login right-pd-input">
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/UiPageValidation/Edit.cshtml:16]: why everywhere form-element-for-login class is used. if it needs to be common name it like form-element-for-common

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/UiPageValidation/Create.cshtml (hunk 7 above)
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/UiPageValidation/Create.cshtml:53]: why it is there 
<script src="~/js/Notification.js"></script>

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Model/Common/UiPageTypeModel.cs (hunk 1 above)
Comment (rajguptaH) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Model/Common/UiPageTypeModel.cs:30]: This Is Needed

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Models/Common/UiPageNavigationDTO.cs [hunk 10]
```diff
@@ -0,0 +1,14 @@
+﻿namespace TestingAndCalibrationLabs.Web.UI.Models
+{
+    public class UiPageNavigationDTO
+    {
+        public int Id { get; set; }
+        public string Url { get; set; }
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Models/Common/UiPageNavigationDTO.cs:6]: Do we need the url property in UI side as well?

**Pull request merged**